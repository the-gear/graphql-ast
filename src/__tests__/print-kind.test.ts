import { readFileSync } from 'fs';
import { parse } from 'graphql';
import { resolve } from 'path';
import printKind from '../print-kind';

const source = readFileSync(resolve(__dirname, './complex.gql'), 'utf8');

describe('print-kind', () => {
  it('works', () => {
    const document = parse(source);
    expect(
      document.definitions.map((def) => `${def.kind} ~> ${printKind(def.kind)}`).join('\n'),
    ).toMatchSnapshot();
  });
});
