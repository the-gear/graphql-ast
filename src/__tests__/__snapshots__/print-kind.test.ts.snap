// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`print-kind works 1`] = `
"ScalarTypeDefinition ~> scalar
DirectiveDefinition ~> directive
ScalarTypeDefinition ~> scalar
InputObjectTypeDefinition ~> input
InterfaceTypeDefinition ~> interface
ObjectTypeDefinition ~> type
ObjectTypeDefinition ~> type
ObjectTypeDefinition ~> type
EnumTypeDefinition ~> enum
EnumTypeExtension ~> extend enum
InterfaceTypeExtension ~> extend interface
ObjectTypeExtension ~> extend type
InputObjectTypeExtension ~> extend input
SchemaDefinition ~> schema
ObjectTypeDefinition ~> type
SchemaExtension ~> extend schema"
`;
